### DISCUSSION TITLE
Cascade ISafeAreaView.IgnoreSafeArea

### DISCUSSION THREAD

**Discussion by DashTheDev** [dotnet/maui#24393] [Q&A] [answered]
For some reason I've not seen any threads related to cascading `ISafeAreaView.IgnoreSafeArea` in layouts. I haven't found a way to set `IgnoreSafeArea` on layouts like `Grid` or `FlexLayout` via a style. The property is never recognised, I assume because the property is not bindable. And I've never seen a cascade property or something similar to `CascadeInputTransparent` but for safe area. 

Maybe I'm missing something, but to get a parent layout and any of its children to ignore safe area I have to do the following.

```XAML
<Grid IgnoreSafeArea="True">

    <Grid IgnoreSafeArea="True">
        <Label Text="Hello!" />
    </Grid>

    <Grid IgnoreSafeArea="True">
        <Label Text="Hello 2!" />
    </Grid>

</Grid>
```

There's got to be a better way of doing this like the ability to set it via a `Style` or a property for cascading the `IgnoreSafeArea` value to child layouts. I don't really want to set the property explicitly on every single layout I have in my view if I need them all to ignore the safe area. I also don't want to use code behind to loop through them all and set them that way. Nor do I want to create something like attached properties to set them and then set that attached property via styling.

Any ideas?

**Comment by LeadAssimilator** [comment, marked as answer]
Report both issues and wait 5+ years for them to get fixed? :stuck_out_tongue:

The IgnoreSafeArea changes have been broken since they were introduced along with Page UseSafeArea. Whomever wrote them was probably drunk at the time.  But to make matters worse the issues have been ignored for nearly 3/4 year now.

I've actually just ran into this myself today and stumbled upon your question while researching solutions.  Short of waiting, adding the property everywhere or building/patching maui yourself, I've come up with 2 tenable solutions:

-  Create your own LayoutHandler implementation to set IgnoreSafeArea = true in SetPlatformView (or ConnectHandler) and register it in your MauiApp builder's ConfigureMauiHandlers for Layout like you would any other handler or compatibility renderer.
-  Create an attached property for IgnoreSafeArea and in App.xaml's resource dictionary, create default styles for TargetType of Grid,AbsoluteLayout,FlexLayout,StackLayout,HorizontalStackLayout, and VerticalStackLayout that each set your attached property to True.

I've tested both options and so far they seem to work fine, but the attached property way would be easily overridable if you need to do in some places.  To do the same with the handler, you would probably have to create an attached property to disable the override which you could check in the handler, but then you might as well just go with the style way.

**Reply by DashTheDev** [reply]
Lol yeah, figured it was something kinda pushed to the side. 

I was definitely thinking the attached property side of things, like you said it's got a bit more flexibility. Good to know it works, I'll implement that and just try to forget about this issue as a whole.

Thanks!
